pkg/arm/data/azuredeploy.json

```diff
@@ -217,139 +143,83 @@
             "properties": {
                 "accountType": "Standard_LRS"
             }
-        },
-        {
-            "type": "Microsoft.Compute/virtualMachineScaleSets",
-            "apiVersion": "2017-12-01",
-            "location": "{{ .ContainerService.Location | js }}",
-            "dependsOn": [
-                "[resourceId('Microsoft.Network/virtualNetworks', 'vnet')]",
-                "[resourceId('Microsoft.Network/networkSecurityGroups', 'nsg-compute')]"
-            ],
-            "name": "ss-compute",
-            "sku": {
-                "tier": "Standard",
-{{- range $app := .ContainerService.Properties.AgentPoolProfiles }}{{ if eq $app.Name `compute` }}
-                "capacity": "{{ $app.Count | js }}",
-                "name": "{{ $app.VMSize | js }}"
-{{- end }}{{ end }}
-            },
-{{- if eq .Config.ImageResourceName "" }}
-            "plan": {
-                "name": "{{ .Config.ImageSKU | js }}",
-                "publisher": "{{ .Config.ImagePublisher | js }}",
-                "product": "{{ .Config.ImageOffer | js }}"
-            },
-{{- end }}
+        }
+{{- range $app := .ContainerService.Properties.AgentPoolProfiles }},
+       {
+            "type": "Microsoft.Network/networkSecurityGroups",
```

Comment:
why are we moving from 2 NSGs to N NSGs based on the number of agent pools?  It seems like duplication when the NSG really is just tied to the role of the agent pool.

Reply:
> For now we can support only 2 compute pool

The agent pools are a slice in the external API PR.  We could validate that you can only have two but we don't restrict it now AFAICS.   In fact, I would suggest that in the (distant?) future more than 2 is probably a normal use case (infra, linux compute, windows compute)

```go
// AgentPoolProfiles represents all the AgentPoolProfiles.
type AgentPoolProfiles []*AgentPoolProfile```

Reply:
This was my intention behind the loop for (distant) distant future. As I understood we cant do more than 3 now because of this https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/azure/azure.go#L95 And I was told azure does not support multiple loadbalancers for scalesets. But original question stays, do you want to have like 2 NSG hardcoded? like `nsg-infa` with `ss-infra` pool and `nsg-compute` with `ss-compute` and future `ss-hpc`,`ss-windows` ?

Reply:
We can always change it later if necessary, leave it like you have it for now.